Source/NNSupportFunctions/arm_nn_mat_mult_nt_t_s8_s32.c

```diff
@@ -54,9 +54,332 @@ arm_cmsis_nn_status arm_nn_mat_mult_nt_t_s8_s32(const int8_t *lhs,
+
+                *dst_ptr = vmladavaq_p_s8(*dst_ptr, v_lhs0, v_rhs0, rmdr);
+
+                dst_ptr += dst_idx_offset;
+                rhs_ptr += rhs_rows;
+            }
+            lhs_ptr += rhs_rows;
+        }
+    }
+
+#elif defined(ARM_MATH_DSP)
+    int16_t lhs_offset_s16 = (int16_t)lhs_offset;
+    const uint32_t lhs_offset_s16x2 = PKHBT(lhs_offset_s16, lhs_offset_s16, 16);
+    for (; rhs_rows_idx >= 8; rhs_rows_idx -= 8)
+    {
+        int32_t *dst_ptr = &dst[0];
+        const int8_t *lhs_ptr = &lhs[0];
+        int32_t lhs_rows_idx = lhs_rows >> 1;
+
+        while (lhs_rows_idx)
+        {
+            const int8_t *rhs_ptr = &rhs[0];
+
+            int32_t lhs000, lhs001, lhs010, lhs011, lhs100, lhs101, lhs110, lhs111;
+            read_pad_and_add(lhs_ptr, &lhs000, &lhs001, lhs_offset_s16x2);
+            read_pad_and_add(&lhs_ptr[4], &lhs010, &lhs011, lhs_offset_s16x2);
+            read_pad_and_add(&lhs_ptr[rhs_rows], &lhs100, &lhs101, lhs_offset_s16x2);
+            read_pad_and_add(&lhs_ptr[rhs_rows + 4], &lhs110, &lhs111, lhs_offset_s16x2);
+
+            for (int32_t rhs_cols_idx = rhs_cols; rhs_cols_idx != 0; rhs_cols_idx--)
+            {
+                int32_t rhs_val00, rhs_val01;
+                read_and_pad(rhs_ptr, &rhs_val00, &rhs_val01);
+
+                dst_ptr[0] = SMLAD(lhs000, rhs_val00, dst_ptr[0]);
+                dst_ptr[0] = SMLAD(lhs001, rhs_val01, dst_ptr[0]);
+                dst_ptr[dst_idx_col_offset] = SMLAD(lhs100, rhs_val00, dst_ptr[dst_idx_col_offset]);
+                dst_ptr[dst_idx_col_offset] = SMLAD(lhs101, rhs_val01, dst_ptr[dst_idx_col_offset]);
+
+                read_and_pad(&rhs_ptr[4], &rhs_val00, &rhs_val01);
+
+                dst_ptr[0] = SMLAD(lhs010, rhs_val00, dst_ptr[0]);
+                dst_ptr[0] = SMLAD(lhs011, rhs_val01, dst_ptr[0]);
+                dst_ptr[dst_idx_col_offset] = SMLAD(lhs110, rhs_val00, dst_ptr[dst_idx_col_offset]);
+                dst_ptr[dst_idx_col_offset] = SMLAD(lhs111, rhs_val01, dst_ptr[dst_idx_col_offset]);
+
+                dst_ptr += dst_idx_offset;
+                rhs_ptr += rhs_rows;
+            }
+            dst_ptr += dst_idx_col_offset;
+
+            lhs_ptr += rhs_rows << 1;
+
+            lhs_rows_idx--;
+        }
+        // Left-over rows
+        if (lhs_rows % 2)
+        {
+            const int8_t *rhs_ptr = &rhs[0];
+            int32_t lhs00, lhs01, lhs10, lhs11;
+            read_pad_and_add(lhs_ptr, &lhs00, &lhs01, lhs_offset_s16x2);
+            read_pad_and_add(&lhs_ptr[4], &lhs10, &lhs11, lhs_offset_s16x2);
+
+            for (int32_t rhs_cols_idx = rhs_cols; rhs_cols_idx != 0; rhs_cols_idx--)
+            {
+                int32_t rhs_val00, rhs_val01, rhs_val10, rhs_val11;
+                read_and_pad(rhs_ptr, &rhs_val00, &rhs_val01);
+                read_and_pad(&rhs_ptr[4], &rhs_val10, &rhs_val11);
+
+                dst_ptr[0] = SMLAD(lhs00, rhs_val00, dst_ptr[0]);
```

Comment:
You could consider doing the operations in this block on a GP register instead of multiple read-writes to memory. Same applies to other placed in this file as well.

Reply:
It looks to me like this is already happening in the disasm